radix-engine/src/system/kernel_modules/transaction_limits/module.rs

```diff
@@ -115,15 +123,38 @@ impl TransactionLimitsModule {
         }
     }
 
-    /// Checks if substate reads count is in the limit.
-    fn validate_substates(&self) -> Result<(), RuntimeError> {
-        if self.substate_store_read > self.limits_config.max_substate_reads {
+    /// Checks if substate reads/writes count and size is in the limit.
+    fn validate_substates(
+        &self,
+        read_size: Option<usize>,
+        write_size: Option<usize>,
+    ) -> Result<(), RuntimeError> {
+        if let Some(size) = read_size {
+            if size > self.limits_config.max_substate_read_size {
+                return Err(RuntimeError::ModuleError(
+                    ModuleError::TransactionLimitsError(
+                        TransactionLimitsError::MaxSubstateReadSizeExceeded(size),
+                    ),
+                ));
+            }
+        }
+        if let Some(size) = write_size {
+            if size > self.limits_config.max_substate_read_size {
```

Comment:
Comparing to wrong parameter.

Reply:
Changed read and write validation to `max_substate_size`, left separate errors for read and write to provide more info to the user.